extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java

```diff
@@ -203,6 +203,19 @@
     @ConfigItem
     public Map<String, SameSiteCookieConfig> sameSiteCookie;
 
+    /**
+     * If responses should be compressed.
+     *
+     * Note that this will attempt to compress all responses, to avoid compressing
+     * already compressed content (such as images) you need to set the following header:
+     * 
+     * Content-Encoding: identity
+     * 
+     * Which will tell vert.x not to compress the response.
+     */
+    @ConfigItem
+    public boolean enableCompression;
```

Comment:
Open question but is it user friendly enough? Could we maybe have rules based on content types?

Reply:
AFAIK this is all that Vert.x provides under the hood. We could provide additional config to install a handler I guess.